rasterio/merge.py

```diff
@@ -141,31 +142,32 @@ def merge(sources, bounds=None, res=None, nodata=None, precision=7):
         # 2. Compute the source window.
         src_window = windows.from_bounds(
             int_w, int_s, int_e, int_n, src.transform,
-            boundless=True, precision=precision)
+            boundless=True)
         logger.debug("Src %s window: %r", src.name, src_window)
 
+        src_window = windows.int_reshape(src_window)
+
         # 3. Compute the destination window.
         dst_window = windows.from_bounds(
             int_w, int_s, int_e, int_n, output_transform,
-            boundless=True, precision=precision)
-
-        logger.debug("Dst window: %r", dst_window)
+            boundless=True)
 
         # 4. Initialize temp array.
         tcount = first.count
-        trows, tcols = tuple(b - a for a, b in dst_window)
+        trows, tcols = (
+            int(round(dst_window.num_rows)), int(round(dst_window.num_cols)))
```

Comment:
is rounding here necessary?  Seems like values have already been rounded by using `windows.int_reshape()`